test/e2e/storage_performant_policy.go

```diff
@@ -0,0 +1,289 @@
+			}
+		})
+
+		g.It("it should not override selinux change policy if pod already has one", func(ctx context.Context) {
+			testContext, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+			pod := getPod(nsObj.Name)
+			recursiveChangePolicy := v1.SELinuxChangePolicyRecursive
+			pod.Spec.SecurityContext.SELinuxChangePolicy = &recursiveChangePolicy
+
+			createdPod, err := kubeClient.CoreV1().Pods(nsObj.Name).Create(testContext, pod, metav1.CreateOptions{})
+			if err != nil {
+				g.Fail(fmt.Sprintf("Failed to create test pod: %v", err))
+			}
+			g.GinkgoLogr.Info("Created pod in namespace", "pod", createdPod.Name, "namespace", nsObj.Name)
+
+			// check if pod has correct selinux change policy
+			runningPod, err := kubeClient.CoreV1().Pods(nsObj.Name).Get(testContext, createdPod.Name, metav1.GetOptions{})
+			if err != nil {
+				g.Fail(fmt.Sprintf("Failed to get pod: %v", err))
+			}
+			if runningPod.Spec.SecurityContext == nil || runningPod.Spec.SecurityContext.SELinuxChangePolicy == nil {
+				g.Fail("SELinuxChangePolicy not set on pod")
+			}
+			if *runningPod.Spec.SecurityContext.SELinuxChangePolicy != v1.SELinuxChangePolicyRecursive {
+				g.Fail(fmt.Sprintf("SELinuxChangePolicy = %v, want %v", *runningPod.Spec.SecurityContext.SELinuxChangePolicy, v1.SELinuxChangePolicyRecursive))
+			}
+		})
+	})
+})
+
+func cleanupNamespace(ctx context.Context, kubeClient kubernetes.Interface, namespace string) {
+	g.GinkgoLogr.Info("Cleaning up namespace", "namespace", namespace)
+	err := kubeClient.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		g.Fail(fmt.Sprintf("Failed to delete test namespace %q: %v", namespace, err))
+	}
+
+	// Wait for the namespace to be deleted
+	err = wait.PollUntilContextCancel(ctx, waitPollInterval, false, func(ctx context.Context) (bool, error) {
+		_, err := kubeClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return true, nil // Namespace deleted successfully
+		}
+		if err != nil {
+			g.GinkgoLogr.Error(err, "Error checking namespace", "namespace", namespace)
+			return false, err // Error occurred, retry
+		}
+		g.GinkgoLogr.Info("Namespace still exists, waiting for deletion", "namespace", namespace)
+		return false, nil // Namespace still exists, keep waiting
+	})
+
+	if err != nil {
+		g.Fail(fmt.Sprintf("Failed to wait for namespace %q deletion: %v", namespace, err))
+	}
+	// Log successful deletion
+	g.GinkgoLogr.Info("Namespace deleted successfully", "namespace", namespace)
+}
+
+func getPod(ns string) *v1.Pod {
+	falseValue := false
+	trueValue := true
+	uidGIDValue := int64(1000)
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "fsgroup-policy-test-pod-",
+			Namespace:    ns,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    "pause",
+					Image:   "k8s.gcr.io/pause:3.2",
```

Comment:
We need to return this image on `cluster-storage-operator-tests-ext images` somehow.